packages/flutter/test/material/dropdown_test.dart

```diff
@@ -1283,4 +1292,234 @@ void main() {
+    ).toList();
+    expect(itemBoxes.length, equals(2));
+
+    // When isDense is true, the button's height is reduced. The menu items'
+    // heights are not.
+    final List<double> itemBoxesHeight = itemBoxes.map<double>((RenderBox box) => box.size.height).toList();
+    final double menuItemHeight = itemBoxesHeight.reduce(math.max);
+    expect(menuItemHeight, greaterThanOrEqualTo(buttonBox.size.height));
+
+    for (RenderBox itemBox in itemBoxes) {
+      expect(itemBox.attached, isTrue);
+      final Offset buttonBoxCenter = buttonBox.size.center(buttonBox.localToGlobal(Offset.zero));
+      final Offset itemBoxCenter = itemBox.size.center(itemBox.localToGlobal(Offset.zero));
+      expect(buttonBoxCenter.dy, equals(itemBoxCenter.dy));
+    }
+  });
+
+  testWidgets('Dropdown button form field changes style when specified', (WidgetTester tester) async {
+    const String value = 'foo';
+    final UniqueKey itemKey = UniqueKey();
+
+    await tester.pumpWidget(
+      TestApp(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: DropdownButtonFormField<String>(
+            value: value,
+            items: <DropdownMenuItem<String>>[
+              DropdownMenuItem<String>(
+                key: itemKey,
+                value: 'foo',
+                child: const Text(value),
+              ),
+            ],
+            isDense: true,
+            onChanged: (_) { },
+            style: const TextStyle(
+              color: Colors.amber,
+              fontSize: 20.0,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    final RichText richText = tester.widget<RichText>(
+      find.descendant(
+        of: find.byKey(itemKey),
+        matching: find.byType(RichText),
+      ),
+    );
+
+    expect(richText.text.style.color, Colors.amber);
+    expect(richText.text.style.fontSize, 20.0);
+  });
+
+  testWidgets('disabledHint displays on empty items or onChanged for form field', (WidgetTester tester) async {
+    final Key buttonKey = UniqueKey();
+
+    Widget build({ List<String> items, ValueChanged<String> onChanged }) => buildFormFrame(
+      items: items,
+      onChanged: onChanged,
+      buttonKey: buttonKey,
+      value: null,
+      hint: const Text('enabled'),
+      disabledHint: const Text('disabled'),
+    );
+
+    // [disabledHint] should display when [items] is null
+    await tester.pumpWidget(build(items: null, onChanged: onChanged));
+    expect(find.text('enabled'), findsNothing);
+    expect(find.text('disabled'), findsOneWidget);
+
+    // [disabledHint] should display when [items] is an empty list.
+    await tester.pumpWidget(build(items: <String>[], onChanged: onChanged));
+    expect(find.text('enabled'), findsNothing);
+    expect(find.text('disabled'), findsOneWidget);
+
+    final RenderBox disabledHintBox = tester.renderObject<RenderBox>(
+      find.byKey(buttonKey),
+    );
+
+    // A Dropdown button with a disabled hint should be the same size as a
```

Comment:
This seems like a separate test. Isolate it out to its own unit test.

Reply:
Done, created separate test.